lib/generators/accessly/install/templates/db/migrate/create_permitted_action_on_objects.rb

```diff
@@ -0,0 +1,14 @@
+class CreatePermittedActionOnObjects < ActiveRecord::Migration<%= migration_version %>
+  def change
+    create_table :accessly_permitted_action_on_objects, force: true, id: :uuid do |t|
+      t.column :segment_id, :integer, default: -1
+      t.column :action, :integer, null: false
+      t.column :actor_id, :integer, null: false
+      t.column :actor_type, :string, null: false
+      t.column :object_type, :string, null: false
+      t.column :object_id, :integer, null: false
+    end
+
+    add_index(:accessly_permitted_action_on_objects, [:segment_id, :action, :actor_id, :actor_type, :object_type, :object_id], unique: true, name: "acessly_paoo_uniq_table_idx")
```

Comment:
### Get a list of users who have permission to edit a particular lesson

#### Why
We might want to show who has permission to edit the lesson so that somebody can possibly choose to revoke somebody's permission.

#### SQL
```sql
SELECT actor_id, actor_type
FROM accessly_permitted_action_on_objects
WHERE segment_id = $1 AND action = $2 AND object_type = $3 AND object_id = $4
```

#### Index that might help
```rb
[:segment_id, :action, :object_type, :object_id]
```

Reply:
@ehourigan I think we should include this index. I believe it will be a common case.

Reply:
we will add this one with :action at the end